collectors/metrics/pkg/metricsclient/metricsclient.go

```diff
@@ -41,42 +41,33 @@ const (
 	maxSeriesLength = 10000
 )
 
-var (
-	gaugeRequestRetrieve = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "metricsclient_request_retrieve",
-		Help: "Tracks the number of metrics retrievals",
-	}, []string{"client", "status_code"})
-	gaugeRequestSend = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "metricsclient_request_send",
-		Help: "Tracks the number of metrics sends",
-	}, []string{"client", "status_code"})
-)
-
-func init() {
-	prometheus.MustRegister(
-		gaugeRequestRetrieve, gaugeRequestSend,
-	)
-}
-
 type Client struct {
 	client      *http.Client
 	maxBytes    int64
 	timeout     time.Duration
 	metricsName string
 	logger      log.Logger
+
+	m *ClientMetrics
+}
+
+type ClientMetrics struct {
+	GaugeRequestRetrieve *prometheus.GaugeVec
+	GaugeRequestSend     *prometheus.GaugeVec
 }
 
 type PartitionedMetrics struct {
 	Families []*clientmodel.MetricFamily
 }
 
-func New(logger log.Logger, client *http.Client, maxBytes int64, timeout time.Duration, metricsName string) *Client {
+func New(logger log.Logger, m *ClientMetrics, client *http.Client, maxBytes int64, timeout time.Duration, metricsName string) *Client {
```

Comment:
Why not rename this `m` param to something descriptive, like `metrics` or `clientMetrics`?